### PR TITLE
add z_stack options 

### PR DESCRIPTION
**Pull request by fdrgsp**
add to the MDA GUI the possibility to select different z_stack mode from `useq` `MDASequence` ( "TopBottom", "RangeAround" and "AboveBelow").

Screen Shot 2021-12-01 at 5 52 29 PM
Screen Shot 2021-12-01 at 5 52 20 PM
Screen Shot 2021-12-01 at 5 52 38 PM

### CODE REVIEW

File: micromanager_gui/multid_widget.py [hunk 1]
```diff
@@ -117,6 +153,45 @@ def _set_enabled(self, enabled: bool):
...
+        self.z_range_abovebelow_doubleSpinBox.setValue(
+            self.above_doubleSpinBox.value() + self.below_doubleSpinBox.value()
+        )
+
+    def _update_n_images(self, value):
+        # when chenging step_size_doubleSpinBox
+        if self.sender().objectName() == "step_size_doubleSpinBox":
+            step = value
+        else:
+            step = self.step_size_doubleSpinBox.value()
```
Comment (tlambert03) [inline micromanager_gui/multid_widget.py:183]: can this just be:  `step = self.step_size_doubleSpinBox.value()`?  is that the intention here?

File: micromanager_gui/multid_widget.py [hunk 2]
```diff
@@ -117,6 +153,45 @@ def _set_enabled(self, enabled: bool):
...
+            f"-{up_down_range} µm <- z -> +{up_down_range} µm"
+        )
+
+    def _update_abovebelow_range(self):
+        self.z_range_abovebelow_doubleSpinBox.setValue(
+            self.above_doubleSpinBox.value() + self.below_doubleSpinBox.value()
+        )
+
+    def _update_n_images(self, value):
+        # when chenging step_size_doubleSpinBox
```
Comment (tlambert03) [inline micromanager_gui/multid_widget.py:179]:
```suggestion
        # when changing step_size_doubleSpinBox
```

File: micromanager_gui/multid_widget.py [hunk 3]
```diff
@@ -117,6 +153,45 @@ def _set_enabled(self, enabled: bool):
...
+    def _set_bottom(self):
+        self.z_bottom_doubleSpinBox.setValue(self._mmc.getZPosition())
+
+    def _update_topbottom_range(self):
+        self.z_range_topbottom_doubleSpinBox.setValue(
+            abs(self.z_top_doubleSpinBox.value() - self.z_bottom_doubleSpinBox.value())
+        )
+
+    def _update_rangearound_label(self):
+        up_down_range = self.zrange_spinBox.value() / 2
```
Comment (tlambert03) [inline micromanager_gui/multid_widget.py:168]: this one could take `value` as the first function parameter, and no longer check `self.zrange_spinBox.value()`

**Review by tlambert03 — APPROVED** [review]
nice!  couple minor comments

**Comment by fdrgsp** [comment]
@tlambert03 fixed

**Pull request merged**